sui_programmability/examples/nfts/sources/Chat.move

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+module NFTs::Chat {
+    use Sui::ID::{Self, ID, VersionedID};
+    use Std::ASCII::{Self, String};
+    use Sui::Transfer;
+    use Sui::TxContext::{Self, TxContext};
+    use Std::Vector::length;
+
+    /// Max text length.
+    const MAX_TEXT_LENGTH: u64 = 512;
+
+    /// Text size overflow.
+    const ETextOverflow: u64 = 0;
+
+    /// Sui Chat NFT (i.e., a post, retweet, like, chat message etc).
+    struct Chat has key, store {
+        id: VersionedID,
+        // The ID of the chat app.
+        app_id: ID,
+        // Post's text.
+        text: String,
+        // Set if referencing an another object (i.e., due to a Like, Retweet, Reply etc).
+        // We allow referencing any object type, not ony Chat NFTs.
+        // By convention applications can use the all zeros address, when no reference is required.
+        ref_id: ID,
```

Comment:
I would suggest `Option<ID>` here. Using an all zero address is effectively introducing a "null" value that clients and callers will need to understand/special-case. I think this is not a pattern we should be promoting.